### PR TITLE
Allow setting email app to use SSL instead of TLS

### PR DESCRIPTION
**Pull request by WoodyWoodsta**
# What this PR does
Adds flexibility of the method of encryption in the SMTP email app. Some email servers are configured to use port 465 (intrinsic TLS) which requires `EMAIL_USE_SSL` instead of `EMAIL_USE_TLS`.

## Which issue(s) this PR fixes
Fixes https://github.com/grafana/oncall/issues/1044

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/base/models/live_setting.py [hunk 1]
```diff
@@ -79,6 +80,7 @@ class LiveSetting(models.Model):
         "EMAIL_HOST_USER": "SMTP server user",
         "EMAIL_HOST_PASSWORD": "SMTP server password",
         "EMAIL_USE_TLS": "SMTP enable/disable TLS",
+        "EMAIL_USE_SSL": "SMTP enable/disable SSL. Should be used mutually excelusively with EMAIL_USE_TLS.",
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:83]:
```suggestion
        "EMAIL_USE_SSL": "SMTP enable/disable SSL. Should be used mutually exclusively with EMAIL_USE_TLS.",
```

File: engine/apps/base/models/live_setting.py
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:1]: can you add a check to [`apps.base.utils.LiveSettingValidator`](https://github.com/grafana/oncall/blob/dev/engine/apps/base/utils.py#L36) to validate that `EMAIL_USE_TLS` and `EMAIL_USE_SSL` are actually mutually exclusive?

File: helm/oncall/templates/_env.tpl
Comment (joeyorlando) [inline helm/oncall/templates/_env.tpl:1]: similar to my other comment, I would suggest some validation on the mutual-exclusivity of `Values.oncall.smtp.tls` and `Values.oncall.smtp.ssl` (a quick search brought up [this](https://stackoverflow.com/a/53516703)).

Also, please add a helm unit test for this.

**Review by joeyorlando — COMMENTED** [review]
@WoodyWoodsta thank you for your contribution! 🙏 

A few small comments. Once those are added I'd be happy to get this approved and merged in :)

File: engine/apps/base/models/live_setting.py
Comment (WoodyWoodsta) [reply engine/apps/base/models/live_setting.py:1]: Ah, I didn't realise that Django enforced the exclusivity. We should definitely do the same. I'll add the checks.

File: helm/oncall/templates/_env.tpl
Comment (WoodyWoodsta) [reply helm/oncall/templates/_env.tpl:1]: https://github.com/grafana/oncall/pull/3911/commits/0f5a1b81485e1a68c29f899a3a1215e06398a39a

File: engine/apps/base/models/live_setting.py
Comment (WoodyWoodsta) [reply engine/apps/base/models/live_setting.py:1]: https://github.com/grafana/oncall/pull/3911/commits/0f5a1b81485e1a68c29f899a3a1215e06398a39a

File: engine/apps/base/models/live_setting.py (hunk 1 above)
Comment (WoodyWoodsta) [reply engine/apps/base/models/live_setting.py]: https://github.com/grafana/oncall/pull/3911/commits/0f5a1b81485e1a68c29f899a3a1215e06398a39a

File: engine/apps/base/models/live_setting.py
Comment (joeyorlando) [reply engine/apps/base/models/live_setting.py:1]: hi 👋 the live setting validation was close but wasn't working 100%. I fixed it in https://github.com/grafana/oncall/pull/3911/commits/406d914f874ce638d7c710cd430493a4d3ac31e0

**Review by joeyorlando — APPROVED** [review]

File: engine/apps/base/models/live_setting.py
Comment (WoodyWoodsta) [reply engine/apps/base/models/live_setting.py:1]: Thanks @joeyorlando - python isn't something I write on a daily basis :D

**Pull request merged**